mbed_greentea/mbed_report_api.py

```diff
@@ -768,6 +770,10 @@ def exporter_memory_metrics_csv(test_result_ext, test_suite_properties=None):
         for test_suite_name in test_results:
             test = test_results[test_suite_name]
 
+            # If test didn't pass, memory stats doesn't make much sense
+            if test['single_test_result'] != TEST_RESULT_OK:
+                continue
```

Comment:
Yeah true that is good candidate where if test fails we would want the results of stack. However for failing tests the memory data can't be truly trusted. If test fails at init_xxx() the data reported by it would smaller. So for reporting we are ignoring the fail tests.

Can take out this check so we get data in either case.